include/wlr/interfaces/wlr_output.h

```diff
@@ -19,12 +19,11 @@ struct wlr_output_impl {
 	bool (*set_mode)(struct wlr_output *output, struct wlr_output_mode *mode);
 	bool (*set_custom_mode)(struct wlr_output *output, int32_t width,
 		int32_t height, int32_t refresh);
-	bool (*set_cursor)(struct wlr_output *output, struct wlr_texture *texture,
-		int32_t scale, enum wl_output_transform transform,
-		int32_t hotspot_x, int32_t hotspot_y, bool update_texture);
-	bool (*move_cursor)(struct wlr_output *output, int x, int y);
 	void (*destroy)(struct wlr_output *output);
 	bool (*attach_render)(struct wlr_output *output, int *buffer_age);
+	bool (*cursor_try_set_size)(struct wlr_output *output, int *x, int *y);
+	bool (*cursor_attach_render)(struct wlr_output *output, int *buffer_age);
+	bool (*cursor_commit)(struct wlr_output *output);
```

Comment:
One thing that bothers me in terms of future-proofing is that we're trying to remove cursors from this interface, yet we're still hardcoding the idea that cursors are special into it. The idea that a cursor plane is special seems rife with legacy potential - I don't see any reason why future graphics architectures would make this call instead of just using generalized planes. What are we to do in that case? Steal a plane and make it the "cursor" plane? Abort and make the compositor show it in software? The idea of cursors at all probably doesn't belong at this level of abstraction. I imagine that ideally the cursorness of a plane becomes an enum value for a bitfield in a general planes API, so that the downstream compositor can search through planes and pick it out as the cursor plane themselves (or this hypothetical cursor helper could do the same). But that's another breaking change we'll face later, so we're designing an API today which we know we'll break later, which is just annoying.

Reply:
I think this new API is designed to be temporary, and will be removed when we start exposing universal planes.

Reply:
Yeah, I know. I just want to be cautious and deliberate when we cause churn for downstream compositors.

Reply:
Yeah, API churn is definitely an issue. But the old cursor code was extremely detrimental towards other work I was doing, particularly `wp_linux_explicit_synchronization` and the backend part of renderer v6. I would've had to hack up the interface even further, and I wasn't keen on adding knowingly broken code.
I feel this is a necessary intermediate step for moving forward.

---

For the DRM backend, universal planes really would make this interface pointless. I personally didn't plan on adding planes support for DRM until after all of this other rendering shit is finished, which at my pace would be *way* off in time. But with emersion doing libliftoff, that could come sooner than I anticipate.

But while I was implementing the Wayland backend part of this, I realised that for it, cursors really are special. They're completely unsynchronized with the main surface (unlike a plane would be), you don't get to move it, and if we wanted to add drag-and-drop support for the Wayland backend, we could even expose drag-icons for it.

We could do something planes-like with `wl_subsurfaces`, but cursors don't fit into that.

Basically, this needs more thought.